### PR TITLE
Add ConfigLoader and TemplatedConfigLoader deprecation notices on documentation.

### PR DESCRIPTION
**Pull request by lrcouto**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Adds deprecation notices to the documentation where ConfigLoader and TemplatedConfigLoader are mentioned, directing the user to the migration guide and instructing them to use OmegaConfigLoader instead.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/configuration/advanced_configuration.md [hunk 1]
```diff
@@ -4,8 +4,52 @@ The documentation on [configuration](./configuration_basics.md) describes how to
 By default, Kedro is set up to use the [ConfigLoader](/kedro.config.ConfigLoader) class. Kedro also provides two additional configuration loaders with more advanced functionality: the [TemplatedConfigLoader](/kedro.config.TemplatedConfigLoader) and the [OmegaConfigLoader](/kedro.config.OmegaConfigLoader).
 Each of these classes are alternatives for the default `ConfigLoader` and have different features. The following sections describe each of these classes and their specific functionality in more detail.
 
+## OmegaConfigLoader
+
+[OmegaConf](https://omegaconf.readthedocs.io/) is a Python library tailored for managing settings. It operates as a YAML-based hierarchical system for organizing configurations and supports merging configurations from various sources.
```
Comment (stichbury) [inline docs/source/configuration/advanced_configuration.md:9]:
```suggestion
[OmegaConf](https://omegaconf.readthedocs.io/) is a Python library designed to handle and manage settings. It serves as a YAML-based hierarchical system to organise configurations, which can be structured to accommodate various sources, allowing you to merge settings from multiple locations.
```

File: docs/source/configuration/advanced_configuration.md [hunk 2]
```diff
@@ -4,8 +4,52 @@ The documentation on [configuration](./configuration_basics.md) describes how to
...
+
+From Kedro 0.18.5 you can use the [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader) which uses `OmegaConf` to load data.
+
+```{note}
+`OmegaConfigLoader` is under active development. It was first available from Kedro 0.18.5 with additional features due in later releases. Let us know if you have any feedback about the `OmegaConfigLoader`.
```
Comment (stichbury) [inline docs/source/configuration/advanced_configuration.md:14]:
```suggestion
`OmegaConfigLoader` is under active development. It is available from Kedro version 0.18.5 with additional features due in later releases. Let us know if you have any feedback about the `OmegaConfigLoader`.
```

Comment (stichbury) [reply docs/source/configuration/advanced_configuration.md:14]: You need to say how to let us know (either add a link to Slack or a mailto link)

File: docs/source/configuration/advanced_configuration.md [hunk 3]
```diff
@@ -4,8 +4,52 @@ The documentation on [configuration](./configuration_basics.md) describes how to
...
+      - [Other configuration files](#other-configuration-files)
+    - [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
+    - [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
+    - [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
+
+
 ## TemplatedConfigLoader
 
+```{note}
+`ConfigLoader` and `TemplatedConfigLoader` have been deprecated since Kedro `0.18.12` and will be removed in Kedro `0.19.0`. Refer to the [Migration guide for config loaders](./config_loader_migration.md) for instructions on how to update your code base to use `OmegaConfigLoader`.
```
Comment (stichbury) [inline docs/source/configuration/advanced_configuration.md:50]:
```suggestion
`ConfigLoader` and `TemplatedConfigLoader` have been deprecated since Kedro `0.18.12` and will be removed in Kedro `0.19.0`. Refer to the [migration guide for config loaders](./config_loader_migration.md) for instructions on how to update your code to use `OmegaConfigLoader`.
```

File: docs/source/configuration/configuration_basics.md [hunk 4]
```diff
@@ -4,11 +4,24 @@ This section contains detailed information about Kedro project configuration, wh
 
 Kedro makes use of a configuration loader to load any project configuration files, and the available configuration loader classes are:
 
+```{note}
+`ConfigLoader` and `TemplatedConfigLoader` have been deprecated since Kedro `0.18.12` and will be removed in Kedro `0.19.0`. Refer to the [Migration guide for config loaders](./config_loader_migration.md) for instructions on how to update your code base to use `OmegaConfigLoader`.
```
Comment (stichbury) [inline docs/source/configuration/configuration_basics.md:8]:
```suggestion
`ConfigLoader` and `TemplatedConfigLoader` have been deprecated since Kedro `0.18.12` and will be removed in Kedro `0.19.0`. Refer to the [migration guide for config loaders](./config_loader_migration.md) for instructions on how to update your code base to use `OmegaConfigLoader`.
```

**Review by stichbury — CHANGES_REQUESTED** [review]
Few changes suggested

File: docs/source/configuration/advanced_configuration.md [hunk 5]
```diff
@@ -4,8 +4,52 @@ The documentation on [configuration](./configuration_basics.md) describes how to
...
+    - [How to ensure non default configuration files get loaded](#how-to-ensure-non-default-configuration-files-get-loaded)
+    - [How to bypass the configuration loading rules](#how-to-bypass-the-configuration-loading-rules)
+    - [How to use Jinja2 syntax in configuration](#how-to-use-jinja2-syntax-in-configuration)
+    - [How to do templating with the `OmegaConfigLoader`](#how-to-do-templating-with-the-omegaconfigloader)
+      - [Parameters](#parameters)
+      - [Catalog](#catalog)
+      - [Other configuration files](#other-configuration-files)
+    - [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
+    - [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
+    - [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
```
Comment (ankatiyar) [inline docs/source/configuration/advanced_configuration.md:44]:
```suggestion
- [How to do templating with the `OmegaConfigLoader`](#how-to-do-templating-with-the-omegaconfigloader)
- [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
- [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
- [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
```

Comment (ankatiyar) [reply docs/source/configuration/advanced_configuration.md:44]: Only link sections relevant to advanced `OmegaConfigLoader` usage, not all subheadings

File: docs/source/configuration/advanced_configuration.md [hunk 6]
```diff
@@ -95,38 +139,27 @@ CONFIG_LOADER_ARGS = {
...
```
Comment (ankatiyar) [inline docs/source/configuration/advanced_configuration.md:162]:
```suggestion
- [How to change which configuration files are loaded](#how-to-change-which-configuration-files-are-loaded)
- [How to ensure non default configuration files get loaded](#how-to-ensure-non-default-configuration-files-get-loaded)
- [How to bypass the configuration loading rules](#how-to-bypass-the-configuration-loading-rules)
- [How to use Jinja2 syntax in configuration](#how-to-use-jinja2-syntax-in-configuration)
- [How to do templating with the `OmegaConfigLoader`](#how-to-do-templating-with-the-omegaconfigloader)
 - [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
- [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
- [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
```

Comment (ankatiyar) [reply docs/source/configuration/advanced_configuration.md:162]: This also is linking to previous sections that we don't need to mention here.

Comment (astrojuanlu) [reply docs/source/configuration/advanced_configuration.md:162]: I suspect this is the Markdown VSCode extension? I've observed similar things locally. Maybe the `toctree` directive from Sphinx would be useful here too.

Comment (lrcouto) [reply docs/source/configuration/advanced_configuration.md]: It was indeed the extension. Should be fixed now.

File: docs/source/configuration/advanced_configuration.md [hunk 7]
```diff
@@ -4,8 +4,38 @@ The documentation on [configuration](./configuration_basics.md) describes how to
...
+### Advanced `OmegaConfigLoader` features
+Some advanced use cases of `OmegaConfigLoader` are listed below:
+- [How to do templating with the `OmegaConfigLoader`](#how-to-do-templating-with-the-omegaconfigloader)
+- [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
+- [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
+- [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
+
 ## TemplatedConfigLoader
 
+```{note}
```
Comment (astrojuanlu) [inline docs/source/configuration/advanced_configuration.md:35]: Can we make this a warning instead?

```suggestion
```{warning}
```

File: docs/source/configuration/advanced_configuration.md [hunk 8]
```diff
@@ -95,38 +125,18 @@ CONFIG_LOADER_ARGS = {
...
 
-* [How to change which configuration files are loaded](#how-to-change-which-configuration-files-are-loaded)
-* [How to ensure non default configuration files get loaded](#how-to-ensure-non-default-configuration-files-get-loaded)
-* [How to bypass the configuration loading rules](#how-to-bypass-the-configuration-loading-rules)
-* [How to use Jinja2 syntax in configuration](#how-to-use-jinja2-syntax-in-configuration)
-* [How to do templating with the `OmegaConfigLoader`](#how-to-do-templating-with-the-omegaconfigloader)
-* [How to use global variables with the `OmegaConfigLoader`](#how-to-use-global-variables-with-the-omegaconfigloader)
-* [How to use resolvers in the `OmegaConfigLoader`](#how-to-use-resolvers-in-the-omegaconfigloader)
-* [How to load credentials through environment variables](#how-to-load-credentials-through-environment-variables)
+- [How to change which configuration files are loaded](#how-to-change-which-configuration-files-are-loaded)
```
Comment (astrojuanlu) [inline docs/source/configuration/advanced_configuration.md:132]: You might want to undo these changes as well, @stichbury do we have a style guide on Markdown? I think we're always using `*` for bulleted lists but correct me if I'm wrong.

File: docs/source/configuration/configuration_basics.md [hunk 9]
```diff
@@ -4,11 +4,24 @@ This section contains detailed information about Kedro project configuration, wh
 
 Kedro makes use of a configuration loader to load any project configuration files, and the available configuration loader classes are:
 
+```{note}
```
Comment (astrojuanlu) [inline docs/source/configuration/configuration_basics.md:7]: This should probably also be a warning

```suggestion
```{warning}
```

File: docs/source/configuration/configuration_basics.md [hunk 10]
```diff
@@ -79,13 +92,22 @@ If you want to change the way configuration is loaded, you can either [customise
...
 This section contains a set of guidance for the most common configuration requirements of standard Kedro projects:
 
-* [How to change the setting for a configuration source folder](#how-to-change-the-setting-for-a-configuration-source-folder)
-* [How to change the configuration source folder at run time](#how-to-change-the-configuration-source-folder-at-runtime)
-* [How to read configuration from a compressed file](#how-to-read-configuration-from-a-compressed-file)
-* [How to access configuration in code](#how-to-access-configuration-in-code)
-* [How to specify additional configuration environments ](#how-to-specify-additional-configuration-environments)
-* [How to change the default overriding environment](#how-to-change-the-default-overriding-environment)
-* [How to use only one configuration environment](#how-to-use-only-one-configuration-environment)
+- [Configuration](#configuration)
```
Comment (astrojuanlu) [inline docs/source/configuration/configuration_basics.md:95]: Probably we should undo these changes too?

**Review by astrojuanlu — COMMENTED** [review]
Left a few comments!

File: docs/source/configuration/advanced_configuration.md (hunk 8 above)
Comment (stichbury) [reply docs/source/configuration/advanced_configuration.md:132]: I've seen both and don't mind either. I use `*` myself.

File: docs/source/configuration/configuration_basics.md [hunk 11]
```diff
@@ -79,13 +92,22 @@ If you want to change the way configuration is loaded, you can either [customise
...
+    * [Configuration file names](#configuration-file-names)
+    * [Configuration patterns](#configuration-patterns)
+  * [How to use Kedro configuration](#how-to-use-kedro-configuration)
+    * [How to change the setting for a configuration source folder](#how-to-change-the-setting-for-a-configuration-source-folder)
+    * [How to change the configuration source folder at runtime](#how-to-change-the-configuration-source-folder-at-runtime)
+    * [How to read configuration from a compressed file](#how-to-read-configuration-from-a-compressed-file)
+    * [How to access configuration in code](#how-to-access-configuration-in-code)
+    * [How to specify additional configuration environments](#how-to-specify-additional-configuration-environments)
+    * [How to change the default overriding environment](#how-to-change-the-default-overriding-environment)
+    * [How to use only one configuration environment](#how-to-use-only-one-configuration-environment)
```
Comment (ankatiyar) [inline docs/source/configuration/configuration_basics.md:110]: There's the same problem with all the subheadings here as well, could you revert to the original list for this part?

Comment (lrcouto) [reply docs/source/configuration/configuration_basics.md]: Reverted both.

**Review by stichbury — APPROVED** [review]
LGTM 🙏

**Review by merelcht — APPROVED** [review]
LGTM! ⭐
